src/js/components/deployment-view/index.tsx

```diff
@@ -50,20 +52,40 @@ class DeploymentView extends React.Component<Props, void> {
   }
 
   public componentWillMount() {
-    const { loadPreviewAndComments, isUserLoggedIn, match: { params: { entityType, token, id } } } = this.props;
+    const {
+      loadPreviewAndComments,
+      isUserLoggedIn,
+      isLoggingIn,
+      match: {
+        params: {
+          entityType,
+          token,
+          id,
+        },
+      },
+    } = this.props;
 
     if (!isEntityType(entityType)) {
       console.error('Unknown preview type!');
       return;
     }
 
-    loadPreviewAndComments(id, entityType, token, isUserLoggedIn);
+    if (
+      isUserLoggedIn ||
+      (!isUserLoggedIn && !isLoggingIn) // A non-Minard user viewing an open deployment
```

Comment:
I understand that you need the preview results to know if an anonymous user is allowed to view the page. You could still wait until it is known if the user is anonymous or not. Then the component wouldn't have to know about the `isLoggingIn` state.

Anyway, I started thinking if it would be worthwhile to try to handle the authorization logic on the routing level. Here's an interesting article about that: https://hackernoon.com/role-based-authorization-in-react-c70bb7641db4 and a related project: https://github.com/mjrussell/redux-auth-wrapper.